File: app/src/main/java/com/highcom/passwordmemo/ui/list/BackgroundColorAdapter.java
package com.highcom.passwordmemo.ui.list;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.highcom.passwordmemo.R;

import java.util.List;

public class BackgroundColorAdapter extends ArrayAdapter<BackgroundColorItem> {

    private int mResource;
    private List<BackgroundColorItem> mItems;
    private LayoutInflater mInflater;

    public BackgroundColorAdapter(@NonNull Context context, int resource, @NonNull List<BackgroundColorItem> objects) {
        super(context, resource, objects);

        mResource = resource;
        mItems = objects;
        mInflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View view;

        if (convertView != null) {
            view = convertView;
        }
        else {
            view = mInflater.inflate(mResource, null);
        }

        // リストビューに表示する要素を取得
        BackgroundColorItem item = mItems.get(position);

        // 背景色と名前を設定
        TextView title = view.findViewById(R.id.backgroundColorName);
        title.setText(item.getColorName());
        LinearLayout linearLayout = view.findViewById(R.id.backgroundColorRow);
        linearLayout.setBackgroundColor(item.getColorCode());

        return view;
    }
}

File: app/src/main/java/com/highcom/passwordmemo/ui/list/BackgroundColorAdapter.kt
package com.highcom.passwordmemo.ui.list

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.highcom.passwordmemo.databinding.RowBackgroundColorBinding

/**
 * 背景色一覧表示用アダプタ
 *
 * @property mResource リソースID
 * @property mItems 背景色一覧データ
 * @constructor
 * 背景色一覧表示用アダプタコンストラクタ
 *
 * @param context コンテキスト
 */
class BackgroundColorAdapter(
    context: Context,
    private val mResource: Int,
    private val mItems: List<BackgroundColorItem>
) : ArrayAdapter<BackgroundColorItem?>(context, mResource, mItems) {
    private val mInflater: LayoutInflater

    init {
        mInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    @SuppressLint("ViewHolder")
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val binding = RowBackgroundColorBinding.inflate(mInflater)
        val view: View = convertView ?: binding.root

        // リストビューに表示する要素を取得
        val item = mItems[position]

        // 背景色と名前を設定
        val title = binding.backgroundColorName
        title.text = item.colorName
        val linearLayout = binding.backgroundColorRow
        item.colorCode?.let { linearLayout.setBackgroundColor(it) }
        return view
    }
}